### PR TITLE
Fix compatibility for memcpy

### PR DESCRIPTION
**Pull request by agorshk**

### CODE REVIEW

**Comment by agorshk** [comment]
make test

File: externals/service_memory.h [hunk 1]
```diff
@@ -137,6 +137,16 @@ void service_memset_seq(T * const ptr, const T value, const size_t num)
...
 
+/**
+ * Saved version of bytes copy between buffers
+ * \param[out] dest               Pointer to new buffer
+ * \param[in]  destSize           Size of new buffer
+ * \param[in]  src                Pointer to source buffer
+ * \param[in]  srcSize            Number of bytes to copy.
+ * \return Status of memory copy, memory copy is successful if zero is returned
+ */
+//int daal_memcpy_s(void *dest, size_t destSize, const void *src, size_t srcSize);
```
Comment (ShvetsKS) [inline externals/service_memory.h]: delete from here comments

File: include/services/daal_memory.h [hunk 2]
```diff
@@ -66,9 +66,21 @@ DAAL_EXPORT void  daal_free(void *ptr);
  * \param[in]  numberOfElements   Size of new buffer
  * \param[in]  src                Pointer to source buffer
  * \param[in]  count              Number of bytes to copy.
- * \return Status of memory copy, memory copy is successful if zero is returned
  */
-DAAL_EXPORT int daal_memcpy_s(void *dest, size_t numberOfElements, const void *src, size_t count);
+DAAL_EXPORT void daal_memcpy_s(void *dest, size_t numberOfElements, const void *src, size_t count);
```
Comment (ShvetsKS) [inline include/services/daal_memory.h]: Mark as deprecated and remove all usages of this function

**Comment by agorshk** [comment]
make test

File: include/services/daal_memory.h (hunk 2 above)
Comment (agorshk) [reply include/services/daal_memory.h]: fixed

File: externals/service_memory.h (hunk 1 above)
Comment (agorshk) [reply externals/service_memory.h]: done

**Comment by agorshk** [comment]
CI: Id_e9f06550-e6d9-f197-8cdc-8cdcd4b723a1

**Comment by agorshk** [comment]
Rerun CI: Id_e9f3d108-ce8f-f1a2-b209-8cdcd4b723a1

**Pull request merged**